src/components/metrics/card/name/index.js

```diff
@@ -1,108 +1,117 @@
 import React, {Component, PropTypes} from 'react'
 
-import TextArea from 'react-textarea-autosize'
+import {EditorState, Editor, ContentState, getDefaultKeyBinding, KeyBindingUtil} from 'draft-js'
 
 import './style.css'
 
+class SimpleEditor extends React.Component {
+  state = {
+    editorState: this._plainTextEditorState(this.props.value)
+  }
+
+  _plainTextEditorState(value) {
+    return EditorState.createWithContent(ContentState.createFromText(value || ''))
+  }
+
+  _onChange(editorState) {
+   return this.setState({editorState})
+  }
+
+  focus() {
+    this.refs.editor.focus()
+  }
+
+  changePlainText(value) {
+    this.setState({editorState: this._plainTextEditorState(value)})
+  }
+
+  getPlainText() {
+    return this.state.editorState.getCurrentContent().getPlainText('')
+  }
+
+  render() {
+    const {editorState} = this.state;
+    return (
+      <div onClick={this.props.isClickable && this.focus.bind(this)}>
+        <Editor
+          editorState={editorState}
+          onBlur={this.props.onBlur}
+          onChange={this._onChange.bind(this)}
+          tabIndex={2}
+          ref='editor'
+          placeholder={this.props.placeholder}
+        />
+      </div>
+    );
+  }
+}
+
 export default class MetricName extends Component {
   displayName: 'MetricName'
 
   static propTypes = {
     name: PropTypes.string,
     inSelectedCell: PropTypes.bool.isRequired,
-    onChange: PropTypes.func.isRequired,
-    editable: PropTypes.bool.isRequired,
+    onChange: PropTypes.func.isRequired
   }
 
   state = {
-    value: this.props.name,
-    editing: false,
-    persistEditing: false,
-  }
-
-  _editable() {
-    if (this.state.persistEditing) {
-      return this.state.editing
-    } else {
-      return this.props.editable && this.state.editing
-    }
+    value: this.props.name
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.name === this.state.value) {this.setState({value: nextProps.name})}
+    if (!nextProps.inSelectedCell && (this.props.name !== nextProps.name) && (nextProps.name !== this.value())) {
```

Comment:
This change causes a bug with undo. To reproduce, change a metric's name, then select the cell (but not the text field) and hit ctrl-z. The name won't change back (but you can tell by the metric flickering that undo is being called).